tests/test_client/abstract_client.rs

```diff
@@ -0,0 +1,329 @@
+// Copyright 2019 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use log::error;
+use parsec_client::auth::AuthenticationData;
+use parsec_client::core::basic_client::BasicClient;
+use parsec_client::error::Result;
+use parsec_interface::operations::list_providers::ProviderInfo;
+use parsec_interface::operations::psa_algorithm::{Algorithm, AsymmetricSignature, Hash};
+use parsec_interface::operations::psa_key_attributes::*;
+use parsec_interface::operations::psa_key_attributes::{KeyAttributes, KeyPolicy, UsageFlags};
+use parsec_interface::requests::{Opcode, ProviderID};
+use std::collections::{HashMap, HashSet};
+
+/// Client structure automatically choosing a provider and high-level operation functions.
+#[derive(Debug)]
+pub struct TestClient {
```

Comment:
About naming, I am thinking: this structure `TestClient` in `test_client/abstract_client.rs` does not feel totally right. Ideally we should keep the same name of the main structure of a file and its name. Any idea?

Suggestion:
* have the folder `tests/test_clients` (note the **ss**)
* inside have `Abstract`, `Stress` and `Raw` structures under `abstract.rs`, `stress.rs` and `raw.rs`. Dropping `test_client` as it is in the upper module's name 💇‍♂️

Reply:
`abstract` is an identifier.

Also, I'm not sure about having the clients named `Abstract`, `Stress` etc. - while it is true that they reside within a `test_clients` module, it's confusing to take `Client` part out of the structure name, as when you want to use it somewhere you shouldn't need to do `test_clients::Stress`; even this looks dubious to me - what's `Stress` exactly? I think the whole idea of namespacing works and is helpful in some circumstances, but is a bit over the top in others. 

I'll rename the modules within `test_clients`, but will keep the `Client` part of the struct name.